src/components/change-view/change-view.jsx

```diff
@@ -43,55 +44,43 @@ const diffSettingsStorage = 'edgi.wm.ui.diff_settings';
  * @param {ChangeViewProps} props
  */
 export default class ChangeView extends React.Component {
+  // TODO: This fires more often than is probably necessary.
+  // see about alternatives to getDerivedStateFromProps
+  // https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html
   static getDerivedStateFromProps (props, state) {
     // Ensure that the current diff type is relevant to the content we are
     // comparing. If not, switch to a relevant type.
     if (props.from && props.to) {
-      const relevantTypes = relevantDiffTypes(props.from, props.to, props.page);
-      if (!relevantTypes.find(type => type.value === state.diffType)) {
-        let diffType = relevantTypes[0].value;
-        if (relevantTypes.find(type => type.value === defaultDiffType)) {
-          diffType = defaultDiffType;
-        }
-        return { diffType };
-      }
+      return {diffType: findDiffType(props.from, props.to, props.page, state.diffType)};
     }
 
     return null;
   }
 
   constructor (props) {
     super(props);
+
+    const page = props.page;
+    const diffType = (page.versions && page.versions.length > 1)
+      ? findDiffType(props.from, props.to, page)
+      : undefined;
```

Comment:
Hmmm, so the main difference between this and `getDerivedStateFromProps` is the check for `page.versions && page.versions.length > 1` rather than `props.from && props.to`. I was having trouble even remembering why these were different and had to go spelunking… it looks like this is a relic from a time when `ChangeView` didn’t get told what versions to show via props, but picked the latest two for the page, and kept track of them in state! See this change way back in 2017: https://github.com/edgi-govdata-archiving/web-monitoring-ui/commit/21431546c1253be59b7a6056193be212cffb968b

Anyway, it looks to me like, nowadays, this is actually the wrong check, and we *should* be checking `props.from && props.to` here instead. But since that means the logic is the same as `getDerivedStateFromProps`, you can just delete this code altogether!

Reply:
great, it's gone!